cluster/juju/layers/kubernetes-e2e/README.md

```diff
@@ -0,0 +1,119 @@
+# Kubernetes End to end
+
+End-to-end (e2e) tests for Kubernetes provide a mechanism to test end-to-end
+behavior of the system, and is the last signal to ensure end user operations
+match developer specifications. Although unit and integration tests provide a
+good signal, in a distributed system like Kubernetes it is not uncommon that a
+minor change may pass all unit and integration tests, but cause unforeseen
+changes at the system level.
+
+The primary objectives of the e2e tests are to ensure a consistent and reliable
+behavior of the kubernetes code base, and to catch hard-to-test bugs before
+users do, when unit and integration tests are insufficient.
+
+
+## Usage
+
+To deploy the e e2e suite, its best to deploy the
+[kubernetes-core bundle](https://github.com/juju-solutions/bundle-kubernetes-core)
+and then relate the `kubernetes-e2e` charm.
+
+```shell
+juju deploy kubernetes-core
+juju deploy kubernetes-e2e
+juju add-relation kubernetes-e2e kubernetes-master
+juju add-relation kubernetes-e2e easyrsa
+```
+
+
+Once the relations have settled, and the `kubernetes-e2e` charm reports
+`Ready to test.` - you may kick off an end to end validation test.
+
+### Running the e2e test
+
+The e2e test is encapsulated as an action to ensure consistent runs of the
+end to end test. The defaults are sensible for most deployments.
+
+```shell
+juju run-action kubernetes-e2e/0 test
+```
+
+### Tuning the e2e test
+
+The e2e test is configurable. By default it will seek to run the gambit of the
```

Comment:
Mention that the test pool is very large and the focus and skip parameters help users run the tests that they are interested in.